File: backend/HomomorphicEncryptionBackend/Model/Stage.py
"""
HomomorphicEncryptionBackend.Model.Stage
"""

__author__ = "VJ Patel (vj@vjpatel.me)"


class Stage:
    """
    Stage
    """

    TYPE_CLIENT = 0
    TYPE_SERVER = 1

    def __init__(self):
        self.__name = None
        self.__type = Stage.TYPE_SERVER
        self.steps = []

    def set_name(self, name):
        """
        Sets the name
        :param str name:
        :return self:
        """
        self.__name = name

        return self

    def get_name(self):
        """
        Returns the name
        :return:
        """
        return self.__name

    def set_type(self, t):
        """
        Sets the type
        :param int type:
        :return self:
        """
        self.__type = t

        return self

    def get_type(self):
        """
        Returns the type
        :return int:
        """
        return self.__type

    def add_step(self, step):
        """
        Adds a Step
        :param Step step:
        :return self:
        """
        self.steps.append(step)

        return self

    def get_steps(self):
        """
        Returns the steps
        :return Step[]:
        """
        return self.steps

    def to_json(self):
        """
        Returns a dict to be serialized into JSON
        :return {}:
        """
        return {
            'name': self.get_name(),
            'type': self.get_type(),
            'steps': self.get_steps()
        }

File: client/src/app/model/stage.ts
import {Step} from "./step";


export class Stage {

  public static HOST_CLIENT = 0;
  public static HOST_SERVER = 1;


  private name: string;
  private host: number;

  private steps: Array<Step>;

  constructor() {
    this.steps = [];
  }

  /**
   * Returns the name
   * @returns {string}
   */
  public getName(): string {
    return this.name;
  }

  /**
   * Sets the name
   * @param name
   * @returns {Stage}
   */
  public setName(name: string): Stage {
    this.name = name;

    return this;
  }

  /**
   * Returns the host
   * @returns {number}
   */
  public getHost(): number {
    return this.host;
  }

  /**
   * Returns whether or not the Stage is executed on the client
   * @returns {boolean}
   */
  public isClient(): boolean {
    return this.host === Stage.HOST_CLIENT;
  }

  /**
   * Returns whether or not the Stage is executed on the server
   * @returns {boolean}
   */
  public isServer(): boolean {
    return this.host === Stage.HOST_SERVER;
  }

  /**
   * Sets the host
   * @param host
   * @returns {Stage}
   */
  public setHost(host: number) {
    this.host = host;

    return this;
  }

  /**
   * Returns the Steps
   * @returns {Array<Step>}
   */
  public getSteps(): Array<Step> {
    return this.steps;
  }

  /**
   * Adds a Step
   * @param step
   * @returns {Stage}
   */
  public addStep(step: Step): Stage {
    this.steps.push(step);

    return this;
  }

}
